client/trino-client/src/main/java/io/trino/client/spooling/encoding/ArrowIpcDataDecoder.java

```diff
@@ -0,0 +1,139 @@
+package io.trino.client.spooling.encoding;
+
+import io.trino.client.Column;
+import io.trino.client.QueryDataDecoder;
+import io.trino.client.ResultRows;
+import io.trino.client.spooling.DataAttributes;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.ipc.ArrowReader;
+import org.apache.arrow.vector.ipc.ArrowStreamReader;
+import org.apache.arrow.vector.util.Text;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+
+public class ArrowIpcDataDecoder implements QueryDataDecoder {
```

Comment:
implemented this to fit with the current row based client and to let me run the existing AbstractTestEngineOnlyQueries tests. not sure if this is actually desirable for arrow data though